In [ ]:
#1. Mount google drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import pandas as pd

ak_60_df = pd.read_csv('/content/drive/MyDrive/devp_2024_test-1_dataset - Sheet1.csv')
ak_60_df.head()

Serial_Id        Order_Id  Order_Date   Ship_Date       Ship_Mode  \
0          1  CA-2016-152156  08-11-2016  11-11-2016    Second Class   
1          2  CA-2016-152156  08-11-2016  11-11-2016    Second Class   
2          3  CA-2016-138688  12-06-2016  16-06-2016    Second Class   
3          4  US-2015-108966  11-10-2015  18-10-2015  Standard Class   
4          5  US-2015-108966  11-10-2015  18-10-2015  Standard Class   

  Business_Segment             City       State Region Product_Category  \
0         Consumer        Henderson    Kentucky  South        Furniture   
1         Consumer        Henderson    Kentucky  South        Furniture   
2        Corporate      Los Angeles  California   West  Office Supplies   
3         Consumer  Fort Lauderdale     Florida  South        Furniture   
4         Consumer  Fort Lauderdale     Florida  South  Office Supplies   

  Product_Sub-Category  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  
0            Bookcases       110.0232        130.9800              2  
1               Chairs       170.7860        243.9800              3  
2               Labels         3.8743          7.3100              2  
3               Tables       268.1217        191.5155              5  
4              Storage         9.9258         11.1840              2

In [ ]:
# Display the Dimensions of Test Data.
print(ak_60_df.shape)

(9994, 14)


In [ ]:
# A3. Create an Unique Sample of 2001 Records using Student Roll Number as Random State.
ak_60_df_unique_sample = ak_60_df.sample(n=2001, random_state=42)

# Display the head of the new sample
display(ak_60_df_unique_sample.head())

# Display the shape of the new sample to confirm the number of records
print(f"Shape of the unique sample: {ak_60_df_unique_sample.shape}")

Serial_Id        Order_Id  Order_Date   Ship_Date       Ship_Mode  \
3125       3126  CA-2015-121720  11-06-2015  12-06-2015     First Class   
1441       1442  CA-2017-128160  19-12-2017  24-12-2017    Second Class   
4510       4511  CA-2016-119935  10-11-2016  14-11-2016  Standard Class   
39           40  CA-2015-117415  27-12-2015  31-12-2015  Standard Class   
4509       4510  CA-2017-167003  23-05-2017  29-05-2017  Standard Class   

     Business_Segment           City       State   Region Product_Category  \
3125        Corporate       Lakeland     Florida    South  Office Supplies   
1441         Consumer  San Francisco  California     West  Office Supplies   
4510        Corporate    Springfield    Missouri  Central        Furniture   
39        Home Office        Houston       Texas  Central        Furniture   
4509         Consumer    Los Angeles  California     West        Furniture   

     Product_Sub-Category  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  
3125              Storage       135.6663         140.952              4  
1441              Binders        12.6060          18.336              2  
4510          Furnishings        10.0710          18.650              2  
39                 Chairs        75.7350          70.686              3  
4509               Tables        59.2371          57.096              3

Shape of the unique sample: (2001, 14)


In [ ]:
# A4. Display the Dimensions of Sample Data.
print(ak_60_df_sample.shape)


(5001, 18)


In [ ]:
# A5. Display the First and Last 03 Records of the Sample Data.
print("First 3 records:")
display(ak_60_df_sample.head(3))

print("\nLast 3 records:")
display(ak_60_df_sample.tail(3))

First 3 records:


Serial_Id        Order_Id Order_Date  Ship_Date       Ship_Mode  \
3948       3949  CA-2016-119963 2016-11-18 2016-11-22  Standard Class   
3368       3369  CA-2015-115511 2015-11-21 2015-11-25  Standard Class   
5526       5527  CA-2017-166933 2017-04-24 2017-04-28  Standard Class   

     Business_Segment           City       State   Region Product_Category  \
3948      Home Office       Pasadena       Texas  Central  Office Supplies   
3368         Consumer      Las Vegas      Nevada     West        Furniture   
5526      Home Office  Santa Barbara  California     West  Office Supplies   

     Product_Sub-Category  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  \
3948               Labels        16.4862          24.424              2   
3368            Bookcases        50.3958          70.980              2   
5526            Fasteners         1.1584           1.810              1   

     Ship_Delay      Cost  Revenue   Profit  
3948     4 days   32.9724   48.848  15.8756  
3368     4 days  100.7916  141.960  41.1684  
5526     4 days    1.1584    1.810   0.6516


Last 3 records:


Serial_Id        Order_Id Order_Date  Ship_Date    Ship_Mode  \
959         960  CA-2015-112571 2015-09-22 2015-09-22     Same Day   
6981       6982  CA-2014-146990 2014-11-07 2014-11-08  First Class   
814         815  CA-2017-143686 2017-05-14 2017-05-14     Same Day   

     Business_Segment           City       State Region Product_Category  \
959          Consumer      Oceanside  California   West        Furniture   
6981        Corporate  New York City    New York   East  Office Supplies   
814          Consumer      Santa Ana  California   West       Technology   

     Product_Sub-Category  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  \
959           Furnishings        75.7020          102.30              2   
6981                Paper         5.4324           10.06              3   
814           Accessories       113.9943          199.99              7   

     Ship_Delay      Cost  Revenue    Profit  
959      0 days  151.4040   204.60   53.1960  
6981     1 days   16.2972    30.18   13.8828  
814      0 days  797.9601  1399.93  601.9699

In [ ]:
# A6. Display Sample Data Information.
ak_60_df_sample.info()

<class 'pandas.core.frame.DataFrame'>
Index: 5001 entries, 3948 to 814
Data columns (total 18 columns):
 #   Column                Non-Null Count  Dtype          
---  ------                --------------  -----          
 0   Serial_Id             5001 non-null   int64          
 1   Order_Id              5001 non-null   object         
 2   Order_Date            5001 non-null   datetime64[ns] 
 3   Ship_Date             5001 non-null   datetime64[ns] 
 4   Ship_Mode             5001 non-null   object         
 5   Business_Segment      5001 non-null   object         
 6   City                  5001 non-null   object         
 7   State                 5001 non-null   object         
 8   Region                5001 non-null   object         
 9   Product_Category      5001 non-null   object         
 10  Product_Sub-Category  5001 non-null   object         
 11  Cost_Per_Unit         5001 non-null   float64        
 12  Price_Per_Unit        5001 non-null   float64        
 13  Quanti

# A7. Brief Note on Sample Data Information.

Based on the sample data information displayed above, here's a brief note:

The sample DataFrame contains 5001 entries and 14 columns. The columns include a mix of data types:
- `int64`: `Serial_Id`, `Quantity_Sold`
- `float64`: `Cost_Per_Unit`, `Price_Per_Unit`
- `object`: `Order_Id`, `Order_Date`, `Ship_Date`, `Ship_Mode`, `Business_Segment`, `City`, `State`, `Region`, `Product_Category`, `Product_Sub-Category`

All columns have 5001 non-null entries, indicating that there are no missing values in this sample dataset. The memory usage of the DataFrame is approximately 715.1 KB.

In [ ]:
# B1. Display the List of Names of the Variables.
print(ak_60_df_sample.columns)

Index(['Serial_Id', 'Order_Id', 'Order_Date', 'Ship_Date', 'Ship_Mode',
       'Business_Segment', 'City', 'State', 'Region', 'Product_Category',
       'Product_Sub-Category', 'Cost_Per_Unit', 'Price_Per_Unit',
       'Quantity_Sold', 'Ship_Delay', 'Cost', 'Revenue', 'Profit'],
      dtype='object')


In [ ]:
# B2. Identify and Display the List of following Variables: Index Variables | Categorical Variables - [Nominal | Ordinal] | Non-Categorical, Non-Index Variables.

# Assuming 'Serial_Id' is the index variable (or could be set as index)
index_variables = ['Serial_Id']

# Categorical variables (object dtype)
categorical_variables = ak_60_df_sample.select_dtypes(include='object').columns.tolist()

# Attempt to classify categorical variables into Nominal and Ordinal
# This is a general classification based on common assumptions and may need adjustment
nominal_variables = []
ordinal_variables = []

for col in categorical_variables:
    # Simple heuristic: if there's a clear order (like ship modes), consider it potentially ordinal
    # Otherwise, consider it nominal. This might need manual adjustment based on data understanding.
    if col in ['Ship_Mode']: # Example: Ship_Mode might have an order (Same Day > First Class > Second Class > Standard Class)
        ordinal_variables.append(col)
    else:
        nominal_variables.append(col)

# Non-Categorical, Non-Index Variables (numerical types)
non_categorical_non_index_variables = ak_60_df_sample.select_dtypes(include=['int64', 'float64']).columns.tolist()

# Remove index variable if it's also a numerical column
if 'Serial_Id' in non_categorical_non_index_variables:
    non_categorical_non_index_variables.remove('Serial_Id')


print("Index Variables:", index_variables)
print("\nCategorical Variables (Nominal):", nominal_variables)
print("\nCategorical Variables (Ordinal - Potential):", ordinal_variables)
print("\nNon-Categorical, Non-Index Variables:", non_categorical_non_index_variables)

Index Variables: ['Serial_Id']

Categorical Variables (Nominal): ['Order_Id', 'Business_Segment', 'City', 'State', 'Region', 'Product_Category', 'Product_Sub-Category']

Categorical Variables (Ordinal - Potential): ['Ship_Mode']

Non-Categorical, Non-Index Variables: ['Cost_Per_Unit', 'Price_Per_Unit', 'Quantity_Sold', 'Cost', 'Revenue', 'Profit']


In [ ]:
# B3. Subset (along with Index) and Display the Categorical Variables.

# Combine index and categorical variables
subset_columns = index_variables + nominal_variables + ordinal_variables

# Create the subset DataFrame
ak_60_df_categorical = ak_60_df_sample[subset_columns]

# Display the head of the subset DataFrame
display(ak_60_df_categorical.head())

# Display the shape of the subset DataFrame
print(f"Shape of the categorical subset: {ak_60_df_categorical.shape}")

Serial_Id        Order_Id Business_Segment           City         State  \
3948       3949  CA-2016-119963      Home Office       Pasadena         Texas   
3368       3369  CA-2015-115511         Consumer      Las Vegas        Nevada   
5526       5527  CA-2017-166933      Home Office  Santa Barbara    California   
7976       7977  US-2015-126753        Corporate   Philadelphia  Pennsylvania   
520         521  CA-2015-157812         Consumer        Houston         Texas   

       Region Product_Category Product_Sub-Category       Ship_Mode  
3948  Central  Office Supplies               Labels  Standard Class  
3368     West        Furniture            Bookcases  Standard Class  
5526     West  Office Supplies            Fasteners  Standard Class  
7976     East  Office Supplies                Paper  Standard Class  
520   Central  Office Supplies              Storage  Standard Class

Shape of the categorical subset: (5001, 9)


In [ ]:
# B4. Subset (along with Index) and Display the Non-Categorical Variables.

# Combine index and non-categorical, non-index variables
subset_columns_non_categorical = index_variables + non_categorical_non_index_variables

# Create the subset DataFrame
ak_60_df_non_categorical = ak_60_df_sample[subset_columns_non_categorical]

# Display the head of the subset DataFrame
display(ak_60_df_non_categorical.head())

# Display the shape of the subset DataFrame
print(f"Shape of the non-categorical subset: {ak_60_df_non_categorical.shape}")

Serial_Id  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold      Cost  \
3948       3949        16.4862          24.424              2   32.9724   
3368       3369        50.3958          70.980              2  100.7916   
5526       5527         1.1584           1.810              1    1.1584   
7976       7977         2.3540           3.424              3    7.0620   
520         521        77.7408          64.784              2  155.4816   

      Revenue   Profit  
3948   48.848  15.8756  
3368  141.960  41.1684  
5526    1.810   0.6516  
7976   10.272   3.2100  
520   129.568 -25.9136

Shape of the non-categorical subset: (5001, 7)


In [ ]:
# B5. Display the Descriptive Statistics of the Categorical Variables.
# Descriptive Stats: Frequency | Relative-Frequency

categorical_cols = nominal_variables + ordinal_variables

for col in categorical_cols:
    print(f"Descriptive Statistics for '{col}':")
    # Frequency
    frequency = ak_60_df_sample[col].value_counts()
    print("\nFrequency:")
    display(frequency)

    # Relative Frequency
    relative_frequency = ak_60_df_sample[col].value_counts(normalize=True)
    print("\nRelative Frequency:")
    display(relative_frequency)

    print("-" * 30) # Separator for better readability

Descriptive Statistics for 'Order_Id':

Frequency:


Order_Id
CA-2017-100111    9
CA-2016-165330    7
US-2016-108504    7
CA-2017-121048    7
CA-2015-104346    6
                 ..
CA-2015-127502    1
CA-2016-135636    1
CA-2017-118668    1
CA-2017-132976    1
CA-2016-158575    1
Name: count, Length: 3347, dtype: int64


Relative Frequency:


Order_Id
CA-2017-100111    0.0018
CA-2016-165330    0.0014
US-2016-108504    0.0014
CA-2017-121048    0.0014
CA-2015-104346    0.0012
                   ...  
CA-2015-127502    0.0002
CA-2016-135636    0.0002
CA-2017-118668    0.0002
CA-2017-132976    0.0002
CA-2016-158575    0.0002
Name: proportion, Length: 3347, dtype: float64

------------------------------
Descriptive Statistics for 'Business_Segment':

Frequency:


Business_Segment
Consumer       2576
Corporate      1511
Home Office     914
Name: count, dtype: int64


Relative Frequency:


Business_Segment
Consumer       0.515097
Corporate      0.302140
Home Office    0.182763
Name: proportion, dtype: float64

------------------------------
Descriptive Statistics for 'City':

Frequency:


City
New York City    458
Los Angeles      387
Philadelphia     261
San Francisco    256
Seattle          219
                ... 
Summerville        1
Missoula           1
Holyoke            1
Kissimmee          1
Manteca            1
Name: count, Length: 468, dtype: int64


Relative Frequency:


City
New York City    0.091582
Los Angeles      0.077385
Philadelphia     0.052190
San Francisco    0.051190
Seattle          0.043791
                   ...   
Summerville      0.000200
Missoula         0.000200
Holyoke          0.000200
Kissimmee        0.000200
Manteca          0.000200
Name: proportion, Length: 468, dtype: float64

------------------------------
Descriptive Statistics for 'State':

Frequency:


State
California              1029
New York                 571
Texas                    467
Pennsylvania             283
Washington               258
Illinois                 248
Ohio                     237
Florida                  211
Michigan                 125
North Carolina           120
Virginia                 109
Arizona                  107
Tennessee                 90
Georgia                   88
Colorado                  83
Indiana                   77
Kentucky                  72
Oregon                    65
Massachusetts             63
Maryland                  60
Wisconsin                 59
New Jersey                57
Delaware                  48
Minnesota                 43
Connecticut               41
Alabama                   35
Arkansas                  35
Nevada                    29
Rhode Island              29
Missouri                  28
Oklahoma                  28
Mississippi               24
Utah                      24
South Carolina            20
Louisiana                 20
New Mexico                18
Nebraska                  16
Iowa                      15
New Hampshire             15
Kansas                     9
Idaho                      9
South Dakota               8
Montana                    7
District of Columbia       7
Vermont                    6
Maine                      3
North Dakota               3
West Virginia              2
Name: count, dtype: int64


Relative Frequency:


State
California              0.205759
New York                0.114177
Texas                   0.093381
Pennsylvania            0.056589
Washington              0.051590
Illinois                0.049590
Ohio                    0.047391
Florida                 0.042192
Michigan                0.024995
North Carolina          0.023995
Virginia                0.021796
Arizona                 0.021396
Tennessee               0.017996
Georgia                 0.017596
Colorado                0.016597
Indiana                 0.015397
Kentucky                0.014397
Oregon                  0.012997
Massachusetts           0.012597
Maryland                0.011998
Wisconsin               0.011798
New Jersey              0.011398
Delaware                0.009598
Minnesota               0.008598
Connecticut             0.008198
Alabama                 0.006999
Arkansas                0.006999
Nevada                  0.005799
Rhode Island            0.005799
Missouri                0.005599
Oklahoma                0.005599
Mississippi             0.004799
Utah                    0.004799
South Carolina          0.003999
Louisiana               0.003999
New Mexico              0.003599
Nebraska                0.003199
Iowa                    0.002999
New Hampshire           0.002999
Kansas                  0.001800
Idaho                   0.001800
South Dakota            0.001600
Montana                 0.001400
District of Columbia    0.001400
Vermont                 0.001200
Maine                   0.000600
North Dakota            0.000600
West Virginia           0.000400
Name: proportion, dtype: float64

------------------------------
Descriptive Statistics for 'Region':

Frequency:


Region
West       1629
East       1422
Central    1126
South       824
Name: count, dtype: int64


Relative Frequency:


Region
West       0.325735
East       0.284343
Central    0.225155
South      0.164767
Name: proportion, dtype: float64

------------------------------
Descriptive Statistics for 'Product_Category':

Frequency:


Product_Category
Office Supplies    2989
Furniture          1092
Technology          920
Name: count, dtype: int64


Relative Frequency:


Product_Category
Office Supplies    0.597680
Furniture          0.218356
Technology         0.183963
Name: proportion, dtype: float64

------------------------------
Descriptive Statistics for 'Product_Sub-Category':

Frequency:


Product_Sub-Category
Binders        740
Paper          684
Furnishings    482
Phones         464
Storage        408
Art            395
Accessories    367
Chairs         330
Appliances     242
Labels         189
Tables         165
Envelopes      128
Bookcases      115
Fasteners      108
Supplies        95
Machines        52
Copiers         37
Name: count, dtype: int64


Relative Frequency:


Product_Sub-Category
Binders        0.147970
Paper          0.136773
Furnishings    0.096381
Phones         0.092781
Storage        0.081584
Art            0.078984
Accessories    0.073385
Chairs         0.065987
Appliances     0.048390
Labels         0.037792
Tables         0.032993
Envelopes      0.025595
Bookcases      0.022995
Fasteners      0.021596
Supplies       0.018996
Machines       0.010398
Copiers        0.007399
Name: proportion, dtype: float64

------------------------------
Descriptive Statistics for 'Ship_Mode':

Frequency:


Ship_Mode
Standard Class    3024
Second Class       975
First Class        739
Same Day           263
Name: count, dtype: int64


Relative Frequency:


Ship_Mode
Standard Class    0.604679
Second Class      0.194961
First Class       0.147770
Same Day          0.052589
Name: proportion, dtype: float64

------------------------------


In [ ]:
# B6. Visualize the Descriptive Statistics of the Categorical Variables.
# Plots: Bar | Pie (Interactive and Exotic)

import plotly.express as px

categorical_cols = nominal_variables + ordinal_variables

for col in categorical_cols:
    print(f"Visualizing '{col}' with Plotly:")

    # Bar Plot (Frequency)
    # Get value counts and reset index, naming the columns explicitly
    freq_df = ak_60_df_sample[col].value_counts().reset_index()
    freq_df.columns = [col, 'Frequency'] # Rename columns for clarity

    fig_bar = px.bar(
        freq_df,
        x='Frequency',
        y=col,
        orientation='h',
        title=f'Frequency of {col}',
        labels={col: col, 'Frequency': 'Frequency'}
    )
    fig_bar.show()

    # Pie Chart (Relative Frequency)
    fig_pie = px.pie(
        ak_60_df_sample,
        names=col,
        title=f'Relative Frequency of {col}',
        hole=0.3 # Add a hole to make it a donut chart for a slightly more "exotic" look
    )
    fig_pie.update_traces(textposition='inside', textinfo='percent+label')
    fig_pie.show()

    print("-" * 30) # Separator for better readability

Visualizing 'Order_Id' with Plotly:


------------------------------
Visualizing 'Business_Segment' with Plotly:


------------------------------
Visualizing 'City' with Plotly:


------------------------------
Visualizing 'State' with Plotly:


------------------------------
Visualizing 'Region' with Plotly:


------------------------------
Visualizing 'Product_Category' with Plotly:


------------------------------
Visualizing 'Product_Sub-Category' with Plotly:


------------------------------
Visualizing 'Ship_Mode' with Plotly:


------------------------------


# B7. Write a brief Inference Note on the Descriptive Statistics of the Categorical Variables.

Based on the descriptive statistics and visualizations of the categorical variables:

*   **Order_Id, Order_Date, Ship_Date:** These appear to be unique identifiers or dates with a large number of distinct values, as expected.
*   **Business_Segment:** The majority of orders come from the 'Consumer' segment, followed by 'Corporate' and 'Home Office'.
*   **City and State:** There is a wide distribution of orders across many cities and states, with some locations like New York City, Los Angeles, and California having a higher frequency.
*   **Region:** The 'West' and 'East' regions have the highest number of orders, followed by 'Central' and 'South'.
*   **Product_Category:** 'Office Supplies' is the most frequent product category, significantly more than 'Furniture' and 'Technology'.
*   **Product_Sub-Category:** 'Binders' and 'Paper' are the most common sub-categories, while 'Machines' and 'Copiers' are the least frequent.
*   **Ship_Mode:** 'Standard Class' is the most frequently used shipping mode, followed by 'Second Class', 'First Class', and 'Same Day'.

Overall, the categorical variable analysis reveals the distribution of orders across different business segments, locations, product categories, sub-categories, and shipping modes in the sample data.

In [ ]:
# B8. Display the Descriptive Statistics of the Non-Categorical Subset.
# Measures of Central Tendency | Measures of Dispersion | Quartiles | Correlation

print("Descriptive Statistics (Central Tendency, Dispersion, Quartiles):")
display(ak_60_df_non_categorical.describe())

print("\nCorrelation Matrix:")
display(ak_60_df_non_categorical.corr())

Descriptive Statistics (Central Tendency, Dispersion, Quartiles):


Serial_Id  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  \
count  5001.000000    5001.000000     5001.000000    5001.000000   
mean   5064.926215      53.908664       61.548237       3.750250   
std    2881.997087     118.985916      138.974529       2.200957   
min       1.000000       0.544500        0.336000       1.000000   
25%    2572.000000       3.541200        5.680000       2.000000   
50%    5118.000000      13.456800       17.430000       3.000000   
75%    7549.000000      58.305600       66.594000       5.000000   
max    9994.000000    2959.992600     3499.990000      14.000000   

               Cost       Revenue       Profit  
count   5001.000000   5001.000000  5001.000000  
mean     201.640801    229.170066    27.529265  
std      499.876776    574.395574   248.374743  
min        0.554400      0.444000 -6599.978000  
25%       13.018500     17.520000     1.748700  
50%       42.619200     56.300000     8.799000  
75%      189.882000    213.080000    29.950000  
max    11839.970400  13999.960000  6719.980800


Correlation Matrix:


Serial_Id  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  \
Serial_Id        1.000000      -0.013530       -0.007396      -0.020319   
Cost_Per_Unit   -0.013530       1.000000        0.898252      -0.002025   
Price_Per_Unit  -0.007396       0.898252        1.000000      -0.005399   
Quantity_Sold   -0.020319      -0.002025       -0.005399       1.000000   
Cost            -0.016415       0.881040        0.783333       0.228109   
Revenue         -0.009418       0.797054        0.882323       0.225040   
Profit           0.011256       0.070106        0.463946       0.061340   

                    Cost   Revenue    Profit  
Serial_Id      -0.016415 -0.009418  0.011256  
Cost_Per_Unit   0.881040  0.797054  0.070106  
Price_Per_Unit  0.783333  0.882323  0.463946  
Quantity_Sold   0.228109  0.225040  0.061340  
Cost            1.000000  0.902244  0.073953  
Revenue         0.902244  1.000000  0.496769  
Profit          0.073953  0.496769  1.000000

In [ ]:
# B9. Visualize the Descriptive Statistics of the Non-Categorical Variables.
# Plots: Scatter | Box-Whiskers | Heatmap (Interactive and Exotic)

import plotly.express as px

# Scatter plot of Cost_Per_Unit vs Price_Per_Unit
fig_scatter = px.scatter(ak_60_df_non_categorical, x='Cost_Per_Unit', y='Price_Per_Unit',
                         title='Scatter Plot of Price vs. Cost Per Unit',
                         template='plotly_dark', # Use a dark template
                         color_discrete_sequence=px.colors.qualitative.Pastel) # Change color scheme
fig_scatter.show()

# Box plot for numerical variables to visualize distribution and outliers
print("\nBox plots for numerical variables:")
for col in non_categorical_non_index_variables:
    fig_box = px.box(ak_60_df_non_categorical, y=col, title=f'Box Plot of {col}',
                     template='plotly_dark', # Use a dark template
                     color_discrete_sequence=['lightblue']) # Change color
    fig_box.show()

# Heatmap of the correlation matrix
print("\nHeatmap of Correlation Matrix:")
fig_heatmap = px.imshow(ak_60_df_non_categorical.corr(), text_auto=True, aspect="auto",
                        title='Correlation Heatmap of Non-Categorical Variables',
                        template='plotly_dark', # Use a dark template
                        color_continuous_scale='Plasma') # Change color scale
fig_heatmap.show()


Box plots for numerical variables:



Heatmap of Correlation Matrix:


# B10. Write a brief Inference Note on the Descriptive Statistics of the Non-Categorical Variables.

Based on the descriptive statistics and visualizations of the non-categorical variables:

*   **Cost_Per_Unit and Price_Per_Unit:** These variables show a strong positive correlation, as expected. The scatter plot confirms this linear relationship. Both variables have a wide range and appear to have some outliers, as indicated by the box plots.
*   **Quantity_Sold:** This variable has a smaller range compared to cost and price. The box plot shows the distribution and potential outliers in the quantity of items sold.
*   **Correlation Matrix:** The heatmap clearly shows the strong positive correlation between 'Cost_Per_Unit' and 'Price_Per_Unit'. 'Serial_Id' shows very weak correlation with the other numerical variables, which is expected as it's an identifier. 'Quantity_Sold' also shows very weak correlation with cost and price in this sample.

Overall, the descriptive statistics and visualizations of the non-categorical variables provide insights into their central tendencies, spread, potential outliers, and the strong relationship between cost and price per unit.

In [ ]:
# C1. Display the Most and Least Preferred Ship_Mode.

# Get the frequency of each Ship_Mode
ship_mode_counts = ak_60_df_sample['Ship_Mode'].value_counts()

# Find the most and least preferred ship modes
most_preferred_ship_mode = ship_mode_counts.idxmax()
least_preferred_ship_mode = ship_mode_counts.idxmin()

print(f"Most Preferred Ship Mode: {most_preferred_ship_mode}")
print(f"Least Preferred Ship Mode: {least_preferred_ship_mode}")

Most Preferred Ship Mode: Standard Class
Least Preferred Ship Mode: Same Day


In [ ]:
# C2. Display Average Delay in Shipment.

# Convert 'Order_Date' and 'Ship_Date' to datetime objects
ak_60_df_sample['Order_Date'] = pd.to_datetime(ak_60_df_sample['Order_Date'], format='%d-%m-%Y')
ak_60_df_sample['Ship_Date'] = pd.to_datetime(ak_60_df_sample['Ship_Date'], format='%d-%m-%Y')

# Calculate the shipment delay
ak_60_df_sample['Ship_Delay'] = ak_60_df_sample['Ship_Date'] - ak_60_df_sample['Order_Date']

# Calculate the average shipment delay
average_ship_delay = ak_60_df_sample['Ship_Delay'].mean()

print(f"Average Shipment Delay: {average_ship_delay}")

Average Shipment Delay: 3 days 23:30:55.068986202


In [ ]:
# C3. Display the Average Variation in Delay in Shipment.

# Calculate the standard deviation of the shipment delay
average_delay_variation = ak_60_df_sample['Ship_Delay'].std()

print(f"Average Variation in Shipment Delay (Standard Deviation): {average_delay_variation}")

Average Variation in Shipment Delay (Standard Deviation): 1 days 17:49:29.001161594


In [ ]:
# C4. Display the Largest and Smallest Business_Segment.

# Get the frequency of each Business_Segment
business_segment_counts = ak_60_df_sample['Business_Segment'].value_counts()

# Find the largest and smallest business segments
largest_business_segment = business_segment_counts.idxmax()
smallest_business_segment = business_segment_counts.idxmin()

print(f"Largest Business Segment: {largest_business_segment}")
print(f"Smallest Business Segment: {smallest_business_segment}")

Largest Business Segment: Consumer
Smallest Business Segment: Home Office


In [ ]:
# C5. Display the Region having Largest Number of States.

# Group by Region and count the number of unique states in each region
states_per_region = ak_60_df_sample.groupby('Region')['State'].nunique()

# Find the region with the largest number of states
region_with_most_states = states_per_region.idxmax()
num_states_in_largest_region = states_per_region.max()

print(f"Region with the largest number of states: {region_with_most_states}")
print(f"Number of states in the {region_with_most_states} region: {num_states_in_largest_region}")

Region with the largest number of states: East
Number of states in the East region: 14


In [ ]:
# C6. Display the Region having Largest Number of States as well as Cities.

# Group by Region and count the number of unique states and cities in each region
region_geo_counts = ak_60_df_sample.groupby('Region').agg(
    num_states=('State', 'nunique'),
    num_cities=('City', 'nunique')
)

# Find the region with the largest number of states
region_with_most_states = region_geo_counts['num_states'].idxmax()
num_states_in_largest_region = region_geo_counts['num_states'].max()

# Find the region with the largest number of cities
region_with_most_cities = region_geo_counts['num_cities'].idxmax()
num_cities_in_largest_region = region_geo_counts['num_cities'].max()


print(f"Region with the largest number of states: {region_with_most_states} ({num_states_in_largest_region} states)")
print(f"Region with the largest number of cities: {region_with_most_cities} ({num_cities_in_largest_region} cities)")

# To find a region that is largest in both, you might need to consider a combined metric
# or if there's one region that's clearly highest in both.
# In this case, we can see from the output which region is largest for each.
# If you need a single "largest in both" based on a combined score, we can calculate that.

Region with the largest number of states: East (14 states)
Region with the largest number of cities: Central (157 cities)


In [ ]:
# C7. Display the Product_Category that faces the Most and Least Delay (on Average) in Shipment.

# Group by Product_Category and calculate the average shipment delay
average_delay_by_category = ak_60_df_sample.groupby('Product_Category')['Ship_Delay'].mean()

# Find the product category with the most and least average delay
most_delayed_category = average_delay_by_category.idxmax()
least_delayed_category = average_delay_by_category.idxmin()

print(f"Product Category with Most Average Delay: {most_delayed_category} (Average Delay: {average_delay_by_category.max()})")
print(f"Product Category with Least Average Delay: {least_delayed_category} (Average Delay: {average_delay_by_category.min()})")

Product Category with Most Average Delay: Office Supplies (Average Delay: 4 days 00:09:38.119772499)
Product Category with Least Average Delay: Technology (Average Delay: 3 days 22:22:57.391304347)


In [ ]:
# C8. Display the Product_Category having Largest Variety of Sub-Category.

# Group by Product_Category and count the number of unique Product_Sub-Categories in each category
sub_categories_per_category = ak_60_df_sample.groupby('Product_Category')['Product_Sub-Category'].nunique()

# Find the Product_Category with the largest number of unique sub-categories
category_with_most_sub_categories = sub_categories_per_category.idxmax()
num_sub_categories_in_largest_category = sub_categories_per_category.max()

print(f"Product Category with the largest variety of sub-categories: {category_with_most_sub_categories}")
print(f"Number of unique sub-categories in the {category_with_most_sub_categories} category: {num_sub_categories_in_largest_category}")

Product Category with the largest variety of sub-categories: Office Supplies
Number of unique sub-categories in the Office Supplies category: 9


In [ ]:
# C9. Display Product-wise Preferred Ship_Mode.

# Group by Product_Sub-Category and find the most frequent Ship_Mode
product_ship_mode = ak_60_df_sample.groupby('Product_Sub-Category')['Ship_Mode'].agg(lambda x: x.mode()[0] if not x.mode().empty else 'No Data')

print("Product-wise Preferred Ship Mode:")
display(product_ship_mode)

Product-wise Preferred Ship Mode:


Product_Sub-Category
Accessories    Standard Class
Appliances     Standard Class
Art            Standard Class
Binders        Standard Class
Bookcases      Standard Class
Chairs         Standard Class
Copiers        Standard Class
Envelopes      Standard Class
Fasteners      Standard Class
Furnishings    Standard Class
Labels         Standard Class
Machines       Standard Class
Paper          Standard Class
Phones         Standard Class
Storage        Standard Class
Supplies       Standard Class
Tables         Standard Class
Name: Ship_Mode, dtype: object

In [ ]:
# C10. Display Top 03 Product_Sub-Category in term of Orders.

# Get the frequency of each Product_Sub-Category
product_sub_category_counts = ak_60_df_sample['Product_Sub-Category'].value_counts()

# Display the top 3 Product_Sub-Categories
top_3_product_sub_categories = product_sub_category_counts.head(3)

print("Top 3 Product Sub-Categories by Number of Orders:")
display(top_3_product_sub_categories)

Top 3 Product Sub-Categories by Number of Orders:


Product_Sub-Category
Binders        740
Paper          684
Furnishings    482
Name: count, dtype: int64

In [ ]:
# D1. Create and Display the following Variables:
# 1. Cost [Quantity_Sold * Cost_Per_Unit].
# 2. Revenue [Quantity_Sold * Price_Per_Unit].
# 3. Profit [Revenue - Cost].

ak_60_df_sample['Cost'] = ak_60_df_sample['Quantity_Sold'] * ak_60_df_sample['Cost_Per_Unit']
ak_60_df_sample['Revenue'] = ak_60_df_sample['Quantity_Sold'] * ak_60_df_sample['Price_Per_Unit']
ak_60_df_sample['Profit'] = ak_60_df_sample['Revenue'] - ak_60_df_sample['Cost']

# Display the head of the DataFrame with the new columns
display(ak_60_df_sample.head())

Serial_Id        Order_Id Order_Date  Ship_Date       Ship_Mode  \
3948       3949  CA-2016-119963 2016-11-18 2016-11-22  Standard Class   
3368       3369  CA-2015-115511 2015-11-21 2015-11-25  Standard Class   
5526       5527  CA-2017-166933 2017-04-24 2017-04-28  Standard Class   
7976       7977  US-2015-126753 2015-08-16 2015-08-20  Standard Class   
520         521  CA-2015-157812 2015-03-22 2015-03-26  Standard Class   

     Business_Segment           City         State   Region Product_Category  \
3948      Home Office       Pasadena         Texas  Central  Office Supplies   
3368         Consumer      Las Vegas        Nevada     West        Furniture   
5526      Home Office  Santa Barbara    California     West  Office Supplies   
7976        Corporate   Philadelphia  Pennsylvania     East  Office Supplies   
520          Consumer        Houston         Texas  Central  Office Supplies   

     Product_Sub-Category  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  \
3948               Labels        16.4862          24.424              2   
3368            Bookcases        50.3958          70.980              2   
5526            Fasteners         1.1584           1.810              1   
7976                Paper         2.3540           3.424              3   
520               Storage        77.7408          64.784              2   

     Ship_Delay      Cost  Revenue   Profit  
3948     4 days   32.9724   48.848  15.8756  
3368     4 days  100.7916  141.960  41.1684  
5526     4 days    1.1584    1.810   0.6516  
7976     4 days    7.0620   10.272   3.2100  
520      4 days  155.4816  129.568 -25.9136

In [ ]:
# D2. Display the Most and Least Profitable Product_Category.

# Group by Product_Category and calculate the total profit for each category
profit_by_category = ak_60_df_sample.groupby('Product_Category')['Profit'].sum()

# Find the product category with the most and least profit
most_profitable_category = profit_by_category.idxmax()
least_profitable_category = profit_by_category.idxmin()

print(f"Most Profitable Product Category: {most_profitable_category} (Total Profit: {profit_by_category.max():.2f})")
print(f"Least Profitable Product Category: {least_profitable_category} (Total Profit: {profit_by_category.min():.2f})")

Most Profitable Product Category: Office Supplies (Total Profit: 72534.24)
Least Profitable Product Category: Furniture (Total Profit: 5077.57)


In [ ]:
# D3. Display the Cross-Tabluation of Aggregated Profit in terms of Business_Segment and Product_Category.

# Group by Business_Segment and Product_Category and calculate the sum of Profit
profit_cross_tab = ak_60_df_sample.groupby(['Business_Segment', 'Product_Category'])['Profit'].sum().unstack()

# Display the cross-tabulation
print("Cross-Tabulation of Aggregated Profit by Business Segment and Product Category:")
display(profit_cross_tab)

Cross-Tabulation of Aggregated Profit by Business Segment and Product Category:


Product_Category  Furniture  Office Supplies  Technology
Business_Segment                                        
Consumer            63.3806       36474.0798  36508.9722
Corporate         3023.8923       23233.1764   9998.6667
Home Office       1990.2958       12826.9858  13554.4052

In [ ]:
# E1. Filter and Display the Sample Dataset based on: Quantity_Sold < 10 and Profit > Median Value of Profit.

# Calculate the median profit
median_profit = ak_60_df_sample['Profit'].median()

# Filter the dataset based on the conditions
filtered_df = ak_60_df_sample[(ak_60_df_sample['Quantity_Sold'] < 10) & (ak_60_df_sample['Profit'] > median_profit)]

# Display the filtered dataset
print("Filtered Dataset (Quantity_Sold < 10 and Profit > Median Profit):")
display(filtered_df.head())
print(f"Shape of the filtered dataset: {filtered_df.shape}")

Filtered Dataset (Quantity_Sold < 10 and Profit > Median Profit):


Serial_Id        Order_Id Order_Date  Ship_Date       Ship_Mode  \
3948       3949  CA-2016-119963 2016-11-18 2016-11-22  Standard Class   
3368       3369  CA-2015-115511 2015-11-21 2015-11-25  Standard Class   
4728       4729  CA-2014-123323 2014-11-07 2014-11-12  Standard Class   
1919       1920  US-2017-111423 2017-08-17 2017-08-19     First Class   
774         775  CA-2017-104220 2017-01-30 2017-02-05  Standard Class   

     Business_Segment           City       State   Region Product_Category  \
3948      Home Office       Pasadena       Texas  Central  Office Supplies   
3368         Consumer      Las Vegas      Nevada     West        Furniture   
4728         Consumer  San Francisco  California     West  Office Supplies   
1919        Corporate      Watertown    New York     East  Office Supplies   
774         Corporate     Des Moines        Iowa  Central  Office Supplies   

     Product_Sub-Category  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  \
3948               Labels        16.4862          24.424              2   
3368            Bookcases        50.3958          70.980              2   
4728              Binders        10.9950          17.592              7   
1919              Binders       118.7094         179.184              5   
774                   Art        14.7095          20.150              2   

     Ship_Delay      Cost  Revenue    Profit  
3948     4 days   32.9724   48.848   15.8756  
3368     4 days  100.7916  141.960   41.1684  
4728     5 days   76.9650  123.144   46.1790  
1919     2 days  593.5470  895.920  302.3730  
774      6 days   29.4190   40.300   10.8810

Shape of the filtered dataset: (2430, 18)


In [ ]:
# E2. Bifurcate and Display the above Filtered Sample Subset into 02 Subsets based on Business_Segment: {Consumer, Corporate}.

# Bifurcate the filtered dataset based on Business_Segment
consumer_subset = filtered_df[filtered_df['Business_Segment'] == 'Consumer']
corporate_subset = filtered_df[filtered_df['Business_Segment'] == 'Corporate']

# Display the head of the Consumer subset
print("Filtered Subset for Consumer Business Segment:")
display(consumer_subset.head())
print(f"Shape of Consumer subset: {consumer_subset.shape}")

print("\nFiltered Subset for Corporate Business Segment:")
display(corporate_subset.head())
print(f"Shape of Corporate subset: {corporate_subset.shape}")

Filtered Subset for Consumer Business Segment:


Serial_Id        Order_Id Order_Date  Ship_Date       Ship_Mode  \
3368       3369  CA-2015-115511 2015-11-21 2015-11-25  Standard Class   
4728       4729  CA-2014-123323 2014-11-07 2014-11-12  Standard Class   
8160       8161  CA-2016-134138 2016-02-16 2016-02-20  Standard Class   
8646       8647  CA-2016-113341 2016-11-25 2016-11-29  Standard Class   
2183       2184  CA-2016-128818 2016-05-07 2016-05-11  Standard Class   

     Business_Segment           City       State Region Product_Category  \
3368         Consumer      Las Vegas      Nevada   West        Furniture   
4728         Consumer  San Francisco  California   West  Office Supplies   
8160         Consumer     Belleville  New Jersey   East        Furniture   
8646         Consumer    Los Angeles  California   West        Furniture   
2183         Consumer  New York City    New York   East       Technology   

     Product_Sub-Category  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  \
3368            Bookcases        50.3958          70.980              2   
4728              Binders        10.9950          17.592              7   
8160               Chairs        95.7432         113.980              2   
8646               Chairs        58.3056          64.784              3   
2183          Accessories        35.5911          39.990              5   

     Ship_Delay      Cost  Revenue   Profit  
3368     4 days  100.7916  141.960  41.1684  
4728     5 days   76.9650  123.144  46.1790  
8160     4 days  191.4864  227.960  36.4736  
8646     4 days  174.9168  194.352  19.4352  
2183     4 days  177.9555  199.950  21.9945

Shape of Consumer subset: (1260, 18)

Filtered Subset for Corporate Business Segment:


Serial_Id        Order_Id Order_Date  Ship_Date       Ship_Mode  \
1919       1920  US-2017-111423 2017-08-17 2017-08-19     First Class   
774         775  CA-2017-104220 2017-01-30 2017-02-05  Standard Class   
8748       8749  CA-2017-128076 2017-02-02 2017-02-07  Standard Class   
2354       2355  CA-2015-111514 2015-08-31 2015-09-02     First Class   
9193       9194  CA-2015-141810 2015-11-02 2015-11-07  Standard Class   

     Business_Segment           City       State   Region Product_Category  \
1919        Corporate      Watertown    New York     East  Office Supplies   
774         Corporate     Des Moines        Iowa  Central  Office Supplies   
8748        Corporate      San Diego  California     West       Technology   
2354        Corporate  San Francisco  California     West  Office Supplies   
9193        Corporate    San Antonio       Texas  Central       Technology   

     Product_Sub-Category  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  \
1919              Binders       118.7094         179.184              5   
774                   Art        14.7095          20.150              2   
8748             Machines       100.4933         119.992              2   
2354              Binders        18.1843          27.448              5   
9193               Phones       152.9624         172.352              2   

     Ship_Delay      Cost  Revenue    Profit  
1919     2 days  593.5470  895.920  302.3730  
774      6 days   29.4190   40.300   10.8810  
8748     5 days  200.9866  239.984   38.9974  
2354     2 days   90.9215  137.240   46.3185  
9193     5 days  305.9248  344.704   38.7792

Shape of Corporate subset: (743, 18)


In [ ]:
# E3. Sort and Display both the Bifurcated Business_Segments Subsets: {Consumer, Corporate}, based on {Product_Category: Ascending, Quantity_Sold: Ascending, Profit: Descending}.

# Sort the Consumer subset
consumer_subset_sorted = consumer_subset.sort_values(by=['Product_Category', 'Quantity_Sold', 'Profit'],
                                                   ascending=[True, True, False])

# Sort the Corporate subset
corporate_subset_sorted = corporate_subset.sort_values(by=['Product_Category', 'Quantity_Sold', 'Profit'],
                                                      ascending=[True, True, False])

# Display the head of the sorted Consumer subset
print("Sorted Consumer Business Segment Subset:")
display(consumer_subset_sorted.head())

print("\nSorted Corporate Business Segment Subset:")
display(corporate_subset_sorted.head())

Sorted Consumer Business Segment Subset:


Serial_Id        Order_Id Order_Date  Ship_Date       Ship_Mode  \
7635       7636  US-2017-158526 2017-12-29 2018-01-01    Second Class   
2013       2014  CA-2016-145905 2016-09-18 2016-09-23  Standard Class   
3473       3474  CA-2017-111815 2017-03-03 2017-03-10  Standard Class   
3868       3869  CA-2015-132486 2015-10-23 2015-10-27    Second Class   
9185       9186  CA-2017-156776 2017-08-07 2017-08-11  Standard Class   

     Business_Segment              City       State   Region Product_Category  \
7635         Consumer        Louisville    Kentucky    South        Furniture   
2013         Consumer            Auburn     Alabama    South        Furniture   
3473         Consumer  Dearborn Heights    Michigan  Central        Furniture   
3868         Consumer         San Diego  California     West        Furniture   
9185         Consumer       Westminster  California     West        Furniture   

     Product_Sub-Category  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  \
7635               Chairs       213.6958         300.980              1   
2013               Chairs       266.7448         350.980              1   
3473               Chairs       133.9252         180.980              1   
3868               Chairs       213.6958         240.784              1   
9185               Chairs       181.2860         207.184              1   

     Ship_Delay      Cost  Revenue   Profit  
7635     3 days  213.6958  300.980  87.2842  
2013     5 days  266.7448  350.980  84.2352  
3473     7 days  133.9252  180.980  47.0548  
3868     4 days  213.6958  240.784  27.0882  
9185     4 days  181.2860  207.184  25.8980


Sorted Corporate Business Segment Subset:


Serial_Id        Order_Id Order_Date  Ship_Date       Ship_Mode  \
3102       3103  CA-2017-119578 2017-12-22 2017-12-27    Second Class   
1841       1842  CA-2016-157763 2016-07-18 2016-07-23  Standard Class   
866         867  CA-2014-149020 2014-01-10 2014-01-15  Standard Class   
4794       4795  CA-2015-144722 2015-03-16 2015-03-23  Standard Class   
7825       7826  CA-2015-127327 2015-02-03 2015-02-07  Standard Class   

     Business_Segment           City         State Region Product_Category  \
3102        Corporate     Providence  Rhode Island   East        Furniture   
1841        Corporate  Bowling Green      Kentucky  South        Furniture   
866         Corporate    Springfield      Virginia  South        Furniture   
4794        Corporate    Los Angeles    California   West        Furniture   
7825        Corporate           Rome      New York   East        Furniture   

     Product_Sub-Category  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  \
3102            Bookcases       170.1546         220.980              1   
1841               Chairs       101.3832         140.810              1   
866           Furnishings        30.6446          51.940              1   
4794          Furnishings        25.0154          43.130              1   
7825               Chairs        75.7350          90.882              1   

     Ship_Delay      Cost  Revenue   Profit  
3102     5 days  170.1546  220.980  50.8254  
1841     5 days  101.3832  140.810  39.4268  
866      5 days   30.6446   51.940  21.2954  
4794     7 days   25.0154   43.130  18.1146  
7825     4 days   75.7350   90.882  15.1470

In [ ]:
# E4. Append and Display the Sorted Business_Segments Subsets: Corporate followed by Consumer.

# Append the sorted Corporate subset followed by the sorted Consumer subset
appended_subsets = pd.concat([corporate_subset_sorted, consumer_subset_sorted])

# Display the head and tail of the appended DataFrame to show the order
print("Appended Business Segments Subsets (Corporate followed by Consumer):")
display(appended_subsets.head())
display(appended_subsets.tail())

print(f"\nShape of the appended dataset: {appended_subsets.shape}")

Appended Business Segments Subsets (Corporate followed by Consumer):


Serial_Id        Order_Id Order_Date  Ship_Date       Ship_Mode  \
3102       3103  CA-2017-119578 2017-12-22 2017-12-27    Second Class   
1841       1842  CA-2016-157763 2016-07-18 2016-07-23  Standard Class   
866         867  CA-2014-149020 2014-01-10 2014-01-15  Standard Class   
4794       4795  CA-2015-144722 2015-03-16 2015-03-23  Standard Class   
7825       7826  CA-2015-127327 2015-02-03 2015-02-07  Standard Class   

     Business_Segment           City         State Region Product_Category  \
3102        Corporate     Providence  Rhode Island   East        Furniture   
1841        Corporate  Bowling Green      Kentucky  South        Furniture   
866         Corporate    Springfield      Virginia  South        Furniture   
4794        Corporate    Los Angeles    California   West        Furniture   
7825        Corporate           Rome      New York   East        Furniture   

     Product_Sub-Category  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  \
3102            Bookcases       170.1546         220.980              1   
1841               Chairs       101.3832         140.810              1   
866           Furnishings        30.6446          51.940              1   
4794          Furnishings        25.0154          43.130              1   
7825               Chairs        75.7350          90.882              1   

     Ship_Delay      Cost  Revenue   Profit  
3102     5 days  170.1546  220.980  50.8254  
1841     5 days  101.3832  140.810  39.4268  
866      5 days   30.6446   51.940  21.2954  
4794     7 days   25.0154   43.130  18.1146  
7825     4 days   75.7350   90.882  15.1470

Serial_Id        Order_Id Order_Date  Ship_Date       Ship_Mode  \
2189       2190  CA-2017-143063 2017-08-10 2017-08-15  Standard Class   
2404       2405  CA-2017-108574 2017-10-07 2017-10-11  Standard Class   
9407       9408  CA-2017-162789 2017-09-22 2017-09-24    Second Class   
1444       1445  CA-2016-133725 2016-05-23 2016-05-28  Standard Class   
9143       9144  US-2017-166688 2017-05-20 2017-05-26  Standard Class   

     Business_Segment           City       State   Region Product_Category  \
2189         Consumer       Columbus     Indiana  Central       Technology   
2404         Consumer    Los Angeles  California     West       Technology   
9407         Consumer  New York City    New York     East       Technology   
1444         Consumer        Chicago    Illinois  Central       Technology   
9143         Consumer      Las Vegas      Nevada     West       Technology   

     Product_Sub-Category  Cost_Per_Unit  Price_Per_Unit  Quantity_Sold  \
2189               Phones       119.5914         161.610              9   
2404          Accessories       101.6718         123.990              9   
9407          Accessories        99.9600         119.000              9   
1444               Phones       203.4926         219.992              9   
9143               Phones       116.9925         124.792              9   

     Ship_Delay       Cost   Revenue    Profit  
2189     5 days  1076.3226  1454.490  378.1674  
2404     4 days   915.0462  1115.910  200.8638  
9407     2 days   899.6400  1071.000  171.3600  
1444     5 days  1831.4334  1979.928  148.4946  
9143     6 days  1052.9325  1123.128   70.1955


Shape of the appended dataset: (2003, 18)


In [ ]:
# E5. Merge and Display the Filtered Business_Segment Subsets: {Consumer, Corporate} in the following Manner:
# 1. For Business_Segment = Consumer, Consider only 02 Variables: {Order_Date, Region}.
# 2. For Business_Segment = Corporate, Consider only 02 Variables: {Order_Date, Product_Category}.
# 3. Merge the 02 Business_Segment Subsets on the Order_Date.

# Select specific columns from the Consumer subset
consumer_subset_selected = consumer_subset[['Order_Date', 'Region']]

# Select specific columns from the Corporate subset
corporate_subset_selected = corporate_subset[['Order_Date', 'Product_Category']]

# Merge the two subsets on 'Order_Date'
# Using an outer merge to keep all dates from both subsets
merged_subsets = pd.merge(consumer_subset_selected, corporate_subset_selected, on='Order_Date', how='outer')

# Display the merged dataset
print("Merged Subsets (Consumer with Region, Corporate with Product_Category, merged on Order_Date):")
display(merged_subsets.head())
print(f"Shape of the merged dataset: {merged_subsets.shape}")

Merged Subsets (Consumer with Region, Corporate with Product_Category, merged on Order_Date):


Order_Date Region Product_Category
0 2014-01-10    NaN        Furniture
1 2014-01-13   East  Office Supplies
2 2014-01-13  South  Office Supplies
3 2014-01-15  South              NaN
4 2014-01-16   East              NaN

Shape of the merged dataset: (1974, 3)


In [ ]:
# E6. Display the Region-wise Count and Product_Category-wise Count from the Merged Subset.

print("Region-wise Count from Merged Subset:")
display(merged_subsets['Region'].value_counts())

print("\nProduct_Category-wise Count from Merged Subset:")
display(merged_subsets['Product_Category'].value_counts())

Region-wise Count from Merged Subset:


Region
West       662
East       433
South      284
Central    252
Name: count, dtype: int64


Product_Category-wise Count from Merged Subset:


Product_Category
Office Supplies    654
Technology         310
Furniture          258
Name: count, dtype: int64

In [ ]:
# E7. Use Sample Dataset to Create and Display the Dictionary of Product_Categories as Keys and the Count of Product_Sub-Categories as Values, as a Dataframe.

# Group by Product_Category and count the number of unique Product_Sub-Categories
sub_category_counts = ak_60_df_sample.groupby('Product_Category')['Product_Sub-Category'].nunique()

# Convert the Series to a dictionary
sub_category_dict = sub_category_counts.to_dict()

# Convert the dictionary to a DataFrame for display
sub_category_df = pd.DataFrame.from_dict(sub_category_dict, orient='index', columns=['Unique_Sub_Category_Count'])

# Display the DataFrame
print("Dictionary of Product Categories and Unique Sub-Category Counts (as DataFrame):")
display(sub_category_df)

Dictionary of Product Categories and Unique Sub-Category Counts (as DataFrame):


Unique_Sub_Category_Count
Furniture                                4
Office Supplies                          9
Technology                               4